src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -395,9 +395,10 @@ internal static string GetRuntimeClassForTypeCreation(IInspectable inspectable,
             return runtimeClassName;
         }
 
-        private static bool ShouldProvideIReference(Type type)
+        private readonly static ConcurrentDictionary<Type, bool> IsWindowsRuntimeTypeCache = new ConcurrentDictionary<Type, bool>();
+        private static bool IsWindowsRuntimeType(Type type)
```

Comment:
This is really IsIReferenceType as it excludes classes.

Reply:
it includes strings, types, and delegates - all reference types